sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -862,6 +862,18 @@ object TypeCoercion extends TypeCoercionBase {
     case _ => None
   }
 
+  // Return whether a string literal can be promoted as the give data type in a binary comparison.
+  private def canPromoteAsInBinaryComparison(dt: DataType) = dt match {
```

Comment:
is it possible to share this code between ansi and non-ansi?

Reply:
Yes and no. Here is for binary comparison, while in ANSI mode it is for binary operators...It is hard to create a function name for this.